buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -160,6 +179,14 @@ ByteBuf allocate(int size, int maxCapacity) {
     private AdaptiveByteBuf allocate(int size, int maxCapacity, Thread currentThread, AdaptiveByteBuf buf) {
         long threadId = currentThread.getId();
         int sizeBucket = AllocationStatistics.sizeBucket(size); // Compute outside of Magazine lock for better ILP.
+        FastThreadLocal<Magazine> fastThreadLocalMagazines = this.fastThreadLocalMagazines;
+        if (fastThreadLocalMagazines != null && currentThread instanceof FastThreadLocalThread) {
+            // TODO this is opinionated: ignoring on purpose non FastThreadLocalThread(s):
```

Comment:
@chrisvest I don't have yet strong opinion on this...

I see 3 options:
1. uses a `MethodHandle::invokeExact` to check for virtual threads current thread: this will help to use `FastThreadLocal`s with no fear as we do for https://github.com/netty/netty/blob/f9e20ac0490d2877c81def47a7d4046e06d9f410/buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java#L516-L524 helping users which relies on something like `io.netty.allocator.useCacheForAllThreads`, to correctly distinguish between event loop threads NOT using `FastThreadLocalThread`, event loop threads extending `FastThreadLocalThread` and non event loop threads still extending `FastThreadLocalThread`
2. leave it as it is, but it means not helping users which are abusing `FastThreadLocalThread`s for non-event lop threads 
3. like 2, but adding a subsequent check (configurable), to disable non-shareable `Magazine`s for `FastThreadLocalThread`s which are not event loop ones; this won't help users which have event loop threads with custom threads

I'm sending a commit for this last option, but feel free to suggest something different.